automates/program_analysis/CAST2GrFN/model/cast_to_air_model.py

```diff
@@ -0,0 +1,417 @@
+from typing import List, Dict
+from enum import Enum
+
+from automates.program_analysis.CAST2GrFN.model.cast import AstNode
+
+
+class CASTToAIRException(Exception):
+    """
+    Used to create exceptions during the CAST to AIR execution
+
+    Args:
+        Exception: An exception that occured during CAST to AIR execution
+    """
+
+    pass
+
+
+class C2AIdentifierType(str, Enum):
+    UNKNOWN = "unknown"
+    VARIABLE = "variable"
+    CONTAINER = "container"
+    LAMBDA = "lambda"
+
+
+class C2AIdentifierInformation(object):
```

Comment:
@dpdicken The definition of this class (and **many** subsequent classes in this file) is very verbose. Is there a reason we can't use Python's [dataclasses](https://docs.python.org/3/library/dataclasses.html) for these class definitions? Please see the definition of `XXIdentifier` classes in `model_assembly/structures.py` for an example of how to use inheritance and dataclasses together.

Reply:
sure, I don't see why we shouldn't use that.